compiler/ast/parser.nim

```diff
@@ -1264,22 +1287,27 @@ proc postExprBlocks(p: var Parser, x: ParsedNode): ParsedNode =
     p.getTok
     p.skipComment(result)
     if p.tok.tokType notin {tkOf, tkElif, tkElse, tkExcept}:
-      var stmtList = p.newNode(nkStmtList, @[parseStmt(p)])
+      var stmtList = p.newNode(pnkStmtList, p.tok, @[parseStmt(p)])
       # to keep backwards compatibility (see tests/vm/tstringnil)
-      if stmtList[0].kind == nkStmtList: stmtList = stmtList[0]
+      if stmtList[0].kind == pnkStmtList: stmtList = stmtList[0]
 
       stmtList.isBlockArg = true
       # stmtList.flags.incl nfBlockArg # XXXX
-      result.add if openingParams.kind == nkEmpty and
-                    openingPragmas.kind == nkEmpty:
+      result.add if openingParams.kind == pnkEmpty and
+                    openingPragmas.kind == pnkEmpty:
                    stmtList
                  else:
                    newProcNode(
-                     nkDo,
-                     stmtList.info,
+                     pnkDo,
+                     p.lex.fileIdx,
+                     stmtList.token,
                      body = stmtList,
-                     params = if openingParams.kind != nkEmpty: openingParams
-                              else: p.newNode(nkFormalParams, @[p.emptyNode]),
+                     params =
+                      if openingParams.kind != pnkEmpty:
+                        openingParams
+                      else:
+                        # TODO: before merge confirm p.tok is correct
+                        p.newNode(pnkFormalParams, p.tok, @[p.emptyNode]),
```

Comment:
Since there are no params in this case there isn't much of a right answer here I think.

Reply:
Yeah, I think there is a convention in place and it needs to become a formalization, presently things are a bit fuzzy when it comes to info/token to fixate on:

- given `foo = bar`, the `pnkAsgn` is at the equals sign, by convention.
- given `foo = 10`, in an enum type definition context, the `pnkEnumFieldDef` is at `foo` or should it be at the assignment, or the tuple?
- given `foo bar`, a `pnkCommand`, in the old parser it's after the `o` in `foo`
- similar to the above, `foo(bar)` a `pnkCall`, uses the `(` position, as that's the "invocation"

Not sure which token/location to fixate on here for the parsed node production, it might be more than one rule but we need something.

Reply:
> given foo = 10, in an enum type definition context, the pnkEnumFieldDef is at foo or should it be at the assignment, or the tuple?

At the assignment would be consistent with how we do it elsewhere and also not redundant with the RHS or LHS location info (or token in this case).

Reply:
In this specific case where there are no formal params but we still generate a formal params node, we can use the token "closest" to where the formal params would be if there were any, so I think `p.tok` before the call to `parseParamList`.

Reply:
`pnkDo` is now sorted, at least.

I'll take a quick peek at the enum field definition.

Reply:
enum is done, too.

Now to figure out this command funny business.

Reply:
OK, line info for `pnkCommand`s are now fixed.

Aside: it's pre-existing, but right now all `pnkCommand` nodes indicated their info based on the position of the first param, where as the call is between the end of call sym and start of parens. Probably need to think about what makes the most sense.

You can see this happening in `tvarargslen.nim`, all the line infos have a column of `9`, except the `pnkCall` which is `8`.